### PR TITLE
Improve group queries

### PR DESCRIPTION
**Pull request by rullzer**
Before we'd also get the diplayname for each group in the backend. In a
separate query. This is of course not ideal as this information is
obtained on each and every query. Now this is queried once and properly
cached.

Also added more caching to the manager.

Signed-off-by: Roeland Jago Douma <roeland@famdouma.nl>

### CODE REVIEW

File: lib/private/Group/Manager.php [hunk 1]
```diff
@@ -172,7 +172,10 @@ public function get($gid) {
 		if (isset($this->cachedGroups[$gid])) {
 			return $this->cachedGroups[$gid];
 		}
-		return $this->getGroupObject($gid);
+		$group = $this->getGroupObject($gid);
+		$this->cachedGroups[$gid] = $group;
```
Comment (kesselb) [inline lib/private/Group/Manager.php:176]: I think getGroupObject does this already.

Comment (rullzer) [reply lib/private/Group/Manager.php:176]: Ah right...

**Comment by MorrisJobke** [comment]
I guess this one here solves already part of the issue: #20815

**Comment by rullzer** [comment]
> I guess this one here solves already part of the issue: #20815

Basically both are improvements. #20815 is making sure we only get what we need.
This one is making sure we get what we need in 1 go if we need it.

**Review by MorrisJobke — APPROVED** [review]
Tested and works 👍

File: lib/private/Group/Database.php [hunk 2]
```diff
@@ -244,15 +247,19 @@ public function getUserGroups($uid) {
...
 			->where($qb->expr()->eq('uid', $qb->createNamedParameter($uid)))
 			->execute();
 
 		$groups = [];
 		while ($row = $cursor->fetch()) {
 			$groups[] = $row['gid'];
-			$this->groupCache[$row['gid']] = $row['gid'];
+			$this->groupCache[$row['gid']] = [
+				'gid' => $row['gid'],
+				'displayname' => $row['displayname'],
```
Comment (ChristophWurst) [inline lib/private/Group/Database.php:261]: it this always `displayname`? I had this issue in Mail recently and I went for `selectAlias` to ensure the database does not return the selected column, like `g.displayname`

Comment (rullzer) [reply lib/private/Group/Database.php:261]: but there is only 1 displayname... so it should not prefix it right?

Comment (ChristophWurst) [reply lib/private/Group/Database.php:261]: I guess so.

**Comment by rullzer** [comment]
One more review please :)

**Review by juliusknorr — APPROVED** [review]

**Comment by juliushaertl** [comment]
php-cs check failure seems unrelated to that change :wink:

**Comment by rullzer** [comment]
I guess a backport makes sense here. At least to stable19 for 19.0.1

**Pull request merged**